apps/go-mod-validator/src/github.ts

```diff
@@ -0,0 +1,114 @@
+import { Octokit } from "octokit";
+
+async function getDefaultBranch(
+  owner: string,
+  repo: string,
+  octokitClient: Octokit,
+): Promise<string> {
+  const repoObject = await octokitClient.request(
+    `GET /repos/${owner}/${repo}/`,
+  );
+  return repoObject.data.default_branch;
+}
+
+async function findCommitInDefaultBranch(
+  owner: string,
+  repo: string,
+  defaultBranch: string,
+  commitSha: string,
+  octokitClient: Octokit,
+) {
```

Comment:
```suggestion
async function isCommitInDefaultBranch(
  owner: string,
  repo: string,
  defaultBranch: string,
  commitSha: string,
  octokitClient: Octokit,
): boolean {
```
naming suggestion to match semantics better